### DISCUSSION TITLE
Stream/Download large (32 GB) XML files

### DISCUSSION THREAD

**Discussion by polymed** [phpro/soap-client#375] [Q&A] [answered]
Hello everyone,

what is the best way to not parse the result but just stream it directly to a file?

Best regards

**Comment by veewee** [comment, marked as answer]
You are most likely better of using an http-client directly for that purpose.
For building the soap request, you might be able to use the `driver::encode()` method:

https://github.com/phpro/soap-client/blob/a91e2feb00391dcc67dc4cd46cca1a652c46c89e/src/Phpro/SoapClient/Soap/Driver/ExtSoap/ExtSoapDriver.php#L78-L81

(or build a raw request)

That way you get a PSR7 HTTP Response, from which you can stream the body resource to a file resource.

**Reply by polymed** [reply]
Thank you very much. This works very well.

Is there also a possibility to use the XML parser (type mapper) afterwards?

**Reply by veewee** [reply]
Decoding only works for full HTTP responses by using the decode() method.
But in that case, you can just use the engine as is and use the http handler.

If you want to parse 32G of XML, I suggest you to use [veewee/xml's Reader component](https://github.com/veewee/xml/blob/master/docs/reader.md). And convert the data you are interested in, into the models you want to use. You can use a tool like jms/serializer for that specific case.
